app/controllers/concerns/foreman/controller/puppet/hosts_controller_extensions.rb

```diff
@@ -0,0 +1,164 @@
+module Foreman::Controller::Puppet::HostsControllerExtensions
+  extend ActiveSupport::Concern
```

Comment:
did you mean to create a separate controller or a mixin? - just by reading the ticket name i read it as a refactor to different controller for puppet actions?